tests/tools/test_parmetis_preproc.py

```diff
@@ -0,0 +1,241 @@
+def test_gen_edge_files(num_chunks, num_parts, edges_fmt, edges_delimiter):
+    """Unit test case for the function
+    tools/distpartitioning/parmetis_preprocess.py::gen_edge_files
+
+    Parameters:
+    -----------
+    num_chunks : int
+        no. of chunks the input graph needs to be partititioned into
+    num_parts : int
+        no. of partitions
+    edges_fmt : string
+        specifying the storage format for the edge files
+    edges_delimiter : string
+        specifying the delimiter used in the edge files
+    """
+    # Create the input dataset
+    with tempfile.TemporaryDirectory() as root_dir:
+
+        # Prepare the state information for firing unit test
+        input_dir = os.path.join(root_dir, "chunked-data")
+        output_dir = os.path.join(root_dir, "preproc_dir")
+
+        # Get the parser object
+        fn_params = namedtuple("fn_params", "input_dir output_dir num_parts")
+        fn_params.input_dir = input_dir
+        fn_params.output_dir = output_dir
+        fn_params.num_parts = num_parts
+
+        # Read the input schema
+        schema_map = _get_test_data(
+            input_dir, num_chunks, edges_fmt, edges_delimiter
+        )
+
+        # Get the global node id offsets for each node type
+        # There is only one node-type in the test graph
+        # which range from 0 thru 9.
+        ntype_gnid_offset = {}
+        ntype_gnid_offset["n1"] = np.array([0, 10 * num_chunks]).reshape(1, 2)
+
+        # Iterate over no. of partitions
+        for rank in range(num_parts):
+
+            # Fire the unit test case and get the results
+            actual_results = gen_edge_files(rank, schema_map, fn_params)
+
+            # Get the gold results for baseline comparision, expected results
+            baseline_results, fmt_results, edge_types = _get_edge_files(
+                schema_map, rank, num_parts
+            )
+
+            # Validate the results with the baseline results
+            # Test 1. no. of files should have the same count per rank
+            assert len(baseline_results) == len(actual_results)
+
+            # Test 2. Check the contents of each file and verify the
+            # file contents match with the expected results.
+            for idx, fname in enumerate(actual_results):
+
+                # Check the file exists
+                assert os.path.isfile(fname)
+                edge_file = fname.split("/")[-1]
+
+                # ``edgetype`` strings are in canonical format,
+                # src_node_type:edge_type:dst_node_type
+                tokens = edge_types[idx].split(":")
+                src_ntype_name = tokens[0]
+                dst_ntype_name = tokens[2]
+
+                # Read both files and compare the edges
+                # Here note that the src and dst end points are global_node_ids
+                target_file = os.path.join(output_dir, f"{edge_file}")
+                target_data = _read_file(target_file, "csv", " ")
```

Comment:
Why csv here? Why not edges_fmt?

Reply:
@classicsong - this is the internal logic of the parmetis_preprocess.py. During the preprocessing stage (that is when we create the edge files and node weight files for parmetis processing) the pipeline always uses csv format to store these files on the disk and with the " " as the delimiter. This is the reason why these values are hard coded in this line.